lib/experimental/add-registered-webfonts-to-theme-json.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Extend theme.json with programmatically registered webfonts.
+ *
+ * @package gutenberg
+ */
+
+if ( ! function_exists( '_wp_add_registered_webfonts_to_theme_json' ) ) {
+	/**
+	 * Add missing fonts data to the global styles.
+	 *
+	 * @param array $data The global styles.
+	 * @return array The global styles with missing fonts data.
+	 */
+	function _wp_add_registered_webfonts_to_theme_json( $data ) {
+		$registered_font_families = wp_webfonts()->get_all_webfonts();
+
+		if ( empty( $registered_font_families ) ) {
+			return $data;
+		}
+
+		// Make sure the path to settings.typography.fontFamilies.theme exists
+		// before adding missing fonts.
+		if ( empty( $data['settings'] ) ) {
+			$data['settings'] = array();
+		}
+		if ( empty( $data['settings']['typography'] ) ) {
+			$data['settings']['typography'] = array();
+		}
+		if ( empty( $data['settings']['typography']['fontFamilies'] ) ) {
+			$data['settings']['typography']['fontFamilies'] = array();
+		}
+
+		/**
```

Comment:
```suggestion
		/*
```

Nit: only docblocks begin with a double `*`, multiline comments only need one.